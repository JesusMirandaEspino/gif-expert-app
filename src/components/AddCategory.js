import React, { useState } from 'react';
import PropTypes  from 'prop-types';

export const AddCategory = ( {  setCategories } ) => {

    // Valor inicial del input
    const [ InputValue, SetInputValue ] = useState( '' );

    // funcion que cambia el valor del input
    const handleInputChange = ( e ) => {
        SetInputValue( e.target.value );

        console.log(  'handleInputChange llamado' );
    }

    // funcion para prevenir el evento normal dal hacer submit del formulario
    const handleSubmit = (e) => {
        e.preventDefault();

        if( InputValue.trim().length > 2 ){
            setCategories( cat => [   InputValue, ...cat  ] );
            SetInputValue('');
        }

        
    }


    return (


                <form className="col-full-12" onSubmit={ handleSubmit } >
                    <label for="category" >Agregar categoria  </label>
                    <input type="text" name="category" id="category" value={ InputValue } onChange={  handleInputChange } />
                </form>


 

    );

}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
