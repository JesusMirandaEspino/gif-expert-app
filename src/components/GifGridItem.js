import React from 'react'

export default function GifGridItem( { id, title, url } ) {



    return (
        <section className="fila" >

            <div className="contenedor1" >

                <div className="col-full-12" >

                    <div className="card  animate__animated animate__fadeIn">
                        <img  src={ url} title={  title }   alt={ title } />
                        <p>{ title }</p> 
                    </div>


                </div>

            </div>
            
        </section  >
    )
}
