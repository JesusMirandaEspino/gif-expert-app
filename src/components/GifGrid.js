import React, { useState, useEffect } from 'react'
import GifGridItem from './GifGridItem';
import { getGisft } from '../helpers/getGif';

export const GifGrid = ({ category }) => {

    const  [ images, setImages ] =  useState([]);

    useEffect( () => {
        getGisft(  category )
        .then( imgs => setImages( imgs )  );
    }, [ ] );



    return (
        <section className="fila">
            <div className="contenedor1" >
                <div className="col-full-12" >
                    <h2> { category } </h2>  
                </div>
                <div className="col-full-12" >
                        {
                            images.map( img => 
                                <h3> <GifGridItem 
                                key={ img.id }  
                                { ...img }/> 
                                </h3>
                            )
                        }
                </div>
            </div>
        </section>
    )
}
