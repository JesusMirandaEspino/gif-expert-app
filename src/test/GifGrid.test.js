import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import {GifGrid} from '../components/GifGrid';


describe( 'Pruebas con <GifGrid />', ()=>{

    const category = 'one punch';


    test('debe de mostrarse correctamente ', () => {




        const wrappers = shallow( <GifGrid category={category}/>  );
        expect( wrappers ).toMatchSnapshot();
        

    });




})


