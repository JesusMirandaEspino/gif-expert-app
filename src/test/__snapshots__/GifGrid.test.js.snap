// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifGrid /> debe de mostrarse correctamente  1`] = `
<section
  className="fila"
>
  <div
    className="contenedor1"
  >
    <div
      className="col-full-12  animate__animated animate__flash"
    >
      <p>
        Loading
      </p>
    </div>
    <div
      className="col-full-12"
    >
      <h2>
         
        one punch
         
      </h2>
    </div>
    <div
      className="col-full-12"
    />
  </div>
</section>
`;
